hack/netpol/pkg/main/main.go

```diff
@@ -0,0 +1,679 @@
+	// better as metrics, obviously, this is only for POC.
+	for _, pod1 := range allPods {
+		for _, pod2 := range allPods {
+			log.Infof("Probing: %s, %s", string(pod1), string(pod2))
+			connected, err := k8s.Probe(pod1.Namespace(), pod1.PodName(), pod2.Namespace(), pod2.PodName(), port)
+			log.Infof("... expected %v , got %v", reachability.Expected.Get(string(pod1), string(pod2)), connected)
+			if err != nil {
+				log.Errorf("unable to make main observation on %s -> %s: %s", string(pod1), string(pod2), err)
+			}
+			reachability.Observe(pod1, pod2, connected)
+			if !connected {
+				if reachability.Expected.Get(string(pod1), string(pod2)) {
+					log.Warnf("FAILED CONNECTION FOR WHITELISTED PODS %s -> %s !!!! ", string(pod1), string(pod2))
+				}
+			}
+		}
+	}
+}
+
+func main() {
+	k8s, err := NewKubernetes()
+	if err != nil {
+		panic(err)
+	}
+	k8s.CleanNetworkPolicies(namespaces)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, TestDefaultDeny)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, TestPodLabelWhitelistingFromBToA)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testInnerNamespaceTraffic)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testEnforcePodAndNSSelector)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testEnforcePodOrNSSelector)
+
+	bootstrap(k8s)
+	testPortsPolicies(k8s)
+
+	// stacked port policies
+	bootstrap(k8s)
+	testWrapperStacked(k8s, testPortsPoliciesStackedOrUpdated, true)
+	// updated port policies
+	bootstrap(k8s)
+	testWrapperStacked(k8s, testPortsPoliciesStackedOrUpdated, false)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testAllowAll)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testNamedPort)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testNamedPortWNamespace)
+
+	bootstrap(k8s)
+	testWrapperPort80(k8s, testEgressOnNamedPort)
+
+	bootstrap(k8s)
+	testWrapperStacked(k8s, TestAllowAllPrecedenceIngress,true )
+
+	/**
+		// TestCIDR
+		TestEgressAndIngressIntegration
+		TestMultipleUpdates(k8s)
+	**/
+}
+
+// testWrapperStaged is for tests which involve steps of mutation.
+type Stack struct {
+	Reachability  *Reachability
+	NetworkPolicy *networkingv1.NetworkPolicy
+	Port          int
+}
+
+// catch all for any type of test, where we use stacks.  these are validated one at a time.
+// probably use this for *all* tests when we port to upstream.
+func testWrapperStacked(k8s *Kubernetes, theTest func(*Kubernetes, bool) (stack []*Stack), stacked bool) {
+	bootstrap(k8s)
+
+	stack := theTest(k8s, stacked)
+	for _, s := range stack {
+		reachability := s.Reachability
+		policy := s.NetworkPolicy
+		if policy != nil {
+			_, err := k8s.CreateOrUpdateNetworkPolicy(policy.Namespace, policy)
+			if err != nil {
+				panic(err)
+			}
+		}
+		validate(k8s, reachability, s.Port)
+		reachability.PrintSummary(true, true, true)
+	}
+}
+
+// For dual port tests... confirms both ports 80 and 81
+func testWrapperPort8081(k8s *Kubernetes, theTest func(k8s *Kubernetes) (*Reachability, *Reachability)) {
```

Comment:
oo these are two ports.. maybe name testWrapperPort80Port81 .. otherwise it sounds like testing on port 8081 ;)

Reply:
also seems like an unused func

Reply:
yup, we dont actually need this wrapper, deleting.